client/asset/interface.go

```diff
@@ -814,6 +822,34 @@ type BotWallet interface {
 	SingleLotRedeemFees(*PreRedeemForm) (uint64, error)
 }
 
+// ShieldedStatus is the balance and addresses associated with the shielded
+// account.
+type ShieldedStatus struct {
+	LastAddress string
+	Balance     uint64
+}
+
+// ShieldedWallet is implemented by ZCash, and enables working with value in
+// shielded pools.
+type ShieldedWallet interface {
+	// ShieldedStatus list the last address and the balance in the shielded
+	// account.
+	ShieldedStatus() (*ShieldedStatus, error)
+	// NewShieldedAddress creates a new shielded address. A shielded address can
+	// be be reused without sacrifice of privacy on-chain, but that doesn't stop
+	// meat-space coordination to reduce privacy.
+	NewShieldedAddress() (string, error)
+	// ShieldFunds moves funds from the transparent account to the shielded
+	// account.
+	ShieldFunds(ctx context.Context, amt uint64) (dex.Bytes, error)
+	// UnshieldFunds moves funds from the shielded account to the transparent
+	// account.
+	UnshieldFunds(ctx context.Context, amt uint64) (dex.Bytes, error)
```

Comment:
Why use `dex.Bytes` for these returns?  We should prefer native types like `[]byte` unless there's a good reason, like if we have a struct field as a `dex.Bytes` to it marshalls to hex instead of base64, but I don't see it on these.